src/components/ErrorBoundary/__tests__/ErrorBoundary-test.js

```diff
@@ -16,41 +16,43 @@ describe('components/ErrorBoundary', () => {
         wrapper.find(WrappedComponent).simulateError(wrappedError);
     };
 
-    describe('should render', () => {
+    describe('render()', () => {
         test('the wrapped component when no error is thrown', () => {
             const wrapper = getWrapper();
 
             expect(wrapper).toMatchSnapshot();
         });
 
-        test('nothing by default when an error is thrown', () => {
+        test('should render default error component an error is thrown', () => {
             const wrapper = getWrapper();
 
             simulateError(wrapper);
 
-            expect(wrapper).toMatchSnapshot();
+            expect(wrapper.find('DefaultError').exists()).toBe(true);
         });
 
         test('the component specified when an error is thrown', () => {
+            const ErrorComponent = () => <div>Error</div>;
             const wrapper = getWrapper({
-                component: <div>Error</div>,
+                errorComponent: ErrorComponent,
             });
 
             simulateError(wrapper);
 
-            expect(wrapper).toMatchSnapshot();
+            expect(wrapper.find(ErrorComponent).exists()).toBe(true);
         });
     });
 
-    describe('onError callback', () => {
-        test('should be be called with any wrapped error', () => {
+    describe('componentDidCatch()', () => {
+        test('should set the error state and call the onError callback', () => {
             const onError = jest.fn();
             const wrapper = getWrapper({
                 onError,
             });
 
             simulateError(wrapper);
 
+            expect(wrapper.state('error')).toBe(wrappedError);
```

Comment:
It doesn't seem ideal to test the internal logic and state of the component. Can we test its inputs/outputs, instead?